### PR TITLE
Select: support keyboard selecting by partially typing label

### LINKED ISSUE #606: Select: support keyboard selecting by partially typing label
**Enhancement** 

Simulate some of the selecting behaviour from the native select box:

* typing characters in to the select when it is closed should select the best matching item
* typing characters in to the select when it is open should highlight the best matching item

### PR DESCRIPTION
**Pull request by matt-gadd**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Resolves #606

### CODE REVIEW

File: src/select/index.ts [hunk 1]
```diff
@@ -94,6 +99,30 @@ export class SelectBase<P extends SelectProperties = SelectProperties> extends T
 		return getOptionValue ? getOptionValue(option, index) === value : option === value;
 	}
 
+	private _getSelectedIndexOnInput(event: KeyboardEvent) {
+		const { key, options = [], getOptionDisabled, getOptionText } = this.properties;
+		if (event.key !== undefined && event.key.length === 1) {
+			clearInterval(this._resetInputTextTimer);
```
Comment (pottedmeat) [inline src/select/index.ts]: `clearTimeout`?

File: src/select/index.ts [hunk 2]
```diff
@@ -94,6 +99,30 @@ export class SelectBase<P extends SelectProperties = SelectProperties> extends T
 		return getOptionValue ? getOptionValue(option, index) === value : option === value;
 	}
 
+	private _getSelectedIndexOnInput(event: KeyboardEvent) {
+		const { key, options = [], getOptionDisabled, getOptionText } = this.properties;
```
Comment (pottedmeat) [inline src/select/index.ts]: Is `key` used in this function?

File: src/select/index.ts (hunk 1 above)
Comment (matt-gadd) [reply src/select/index.ts]: good catch, updated 👍

File: src/select/index.ts (hunk 2 above)
Comment (matt-gadd) [reply src/select/index.ts]: nope, leftover 😄. removed.

**Pull request merged**